src/Microsoft.PowerShell.Commands.Utility/commands/utility/CSVCommands.cs

```diff
@@ -206,8 +207,7 @@ public SwitchParameter NoClobber
         /// Encoding optional flag
         /// </summary>
         [Parameter()]
-        [ValidateSetAttribute(new string[] { "Unicode", "UTF7", "UTF8", "ASCII", "UTF32", "BigEndianUnicode", "Default", "OEM" })]
-        public string Encoding { get; set; }
+        public FileEncoding Encoding { get; set; } = FileEncoding.Unknown;
```

Comment:
We could use #3784 - dynamic validate set, don't expose `FileEncoding` and  allow enhancement to use legacy charsets.